examples/optional/optional.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+
+	"go.uber.org/dig"
+)
+
+// type1 is a "required" dig type by default
+type type1 struct{}
+
+// type2 embeds a dig.Optional object to signal it's ok to ignore in params
+type type2 struct {
+	dig.Optional
```

Comment:
This would work if struct definition is in control of module owners, but providing this struct with `dig.Optional` will be complicated since modules will not have control over where the objects coming from. For example, even `zap.Logger` can be optional in http module, and switched to std logger if it is not provided.